source/slang/slang.cpp

```diff
@@ -293,9 +293,12 @@ int CompileRequest::executeActionsInner()
         if (mSink.GetErrorCount() != 0)
             return 1;
 
-        // Generate initial IR for all the translation
-        // units, if we are in a mode where IR is called for.
-        generateIR();
+        if ((compileFlags & SLANG_COMPILE_FLAG_NO_CODEGEN) == 0)
+        {
```

Comment:
This is fine for now, but I think that eventually we need/want to generate the “initial” IR even if target code generation is disabled. That is, the output of the first compilation phase should be IR + reflection, which feeds into the second phase.